src/SettingsMigrator/Migrator.cs

```diff
@@ -153,65 +152,65 @@ private static void MigrateTestSettings(string oldTestSettingsPath, string newRu
     /// <param name="runSettingsXmlDoc">Runsettings Xml</param>
     private static void MigrateTestSettingsNodesToRunSettings(string testSettingsPath, XmlDocument runSettingsXmlDoc)
     {
-        var testSettingsNodes = ReadTestSettingsNodes(testSettingsPath);
+        TestSettingsNodes testSettingsNodes = ReadTestSettingsNodes(testSettingsPath);
 
         string? testTimeout = null;
-        if (testSettingsNodes.Timeout != null && testSettingsNodes.Timeout.Attributes[TestTimeoutAttributeName] != null)
+        if (testSettingsNodes?.Timeout?.Attributes?[TestTimeoutAttributeName] != null)
         {
-            testTimeout = testSettingsNodes.Timeout.Attributes[TestTimeoutAttributeName].Value;
+            testTimeout = testSettingsNodes?.Timeout?.Attributes?[TestTimeoutAttributeName]?.Value;
         }
 
         string? runTimeout = null;
-        if (testSettingsNodes.Timeout != null && testSettingsNodes.Timeout.Attributes[RunTimeoutAttributeName] != null)
+        if (testSettingsNodes?.Timeout?.Attributes?[RunTimeoutAttributeName] != null)
```

Comment:
According to nullability above, `testSettingsNodes`  cannot be null

```suggestion
        if (testSettingsNodes.Timeout?.Attributes?[RunTimeoutAttributeName] != null)
```